File: src/Eto.Forms.Templates/content/App-VisualBasic/EtoApp.1/MainForm.jeto.vb
#If (UseJeto)
Imports Eto.Forms
Imports Eto.Drawing
Imports Eto.Serialization.Json

Public Class MainForm
	Inherits Form

	Public Sub New()
		JsonReader.Load(Me)
	End Sub
#If IsForm

	Protected Sub HandleClickMe(sender As Object, e As EventArgs)
		MessageBox.Show("I was clicked!")
	End Sub

	Protected Sub HandleAbout(sender As Object, e As EventArgs)
		Dim dlg As New AboutDialog()
		dlg.ShowDialog(Me)
	End Sub

	Protected Sub HandleQuit(sender As Object, e As EventArgs)
		Application.Instance.Quit()
	End Sub
#End If

End Class
#End If

File: src/Eto.Forms.Templates/content/File-CSharp/MainForm.jeto.cs
#if (UseJeto)
using System;
using System.Collections.Generic;
using Eto.Forms;
using Eto.Drawing;
using Eto.Serialization.Json;

namespace EtoApp._1
{	
	public class MainForm : Form
	{	
		public MainForm()
		{
			JsonReader.Load(this);
		}
#if IsForm

		protected void HandleClickMe(object sender, EventArgs e)
		{
			MessageBox.Show("I was clicked!");
		}

		protected void HandleAbout(object sender, EventArgs e)
		{
			new AboutDialog().ShowDialog(this);
		}

		protected void HandleQuit(object sender, EventArgs e)
		{
			Application.Instance.Quit();
		}
#endif
	}
}
#endif